core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -520,7 +520,7 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         |                              (Default: client).
         |  --class CLASS_NAME          Your application's main class (for Java / Scala apps).
         |  --name NAME                 A name of your application.
-        |  --jars JARS                 Comma-separated list of local jars to include on the driver
```

Comment:
Thanks @cloud-fan for your review. Supporting remote jar has been addressed in #18078 , but it still left some codes and docs should be updated, also it doesn't support downloading jars from HTTP(S), FTP, here also adding this support.

So basically this PR address the left problem of #18078 and add glob support.